client/state/data-getters/index.js

```diff
@@ -82,6 +82,47 @@ export const requestGeoLocation = () =>
 		{ fromApi: () => ( { body: { country_short } } ) => [ [ 'geo', country_short ] ] }
 	);
 
+export const requestRecentPostViews = ( siteId, postIds ) => {
+	const key = `recent-post-views-${ siteId }`;
+	const freshness = 5 * 60 * 1000; // 5 min
+
+	// Don't re-request recent post views that are cached and up to date OR
+	// have a pending request.
+	const freshPostIds = postIds.filter(
+		postId =>
+			getHttpData( `${ key }-${ postId }` ).lastUpdated >= Date.now() - freshness &&
+			getHttpData( `${ key }-${ postId }` ).status !== 'pending'
+	);
+	const requestPostIds = difference( postIds, freshPostIds );
+
+	// Break post_ids into chunks of 100 because `stats/views/posts`
+	// is limited to 100 post_ids per query.
+	return chunk( requestPostIds, 100 ).map( postIdsChunk => {
+		const postIdsList = postIdsChunk.join( ',' );
+
+		return requestHttpData(
+			`${ key }-${ postIdsList }`,
+			http(
+				{
+					method: 'GET',
+					path: `/sites/${ siteId }/stats/views/posts`,
+					apiVersion: '1.1',
+					query: {
+						post_ids: postIdsList,
+						num: 30,
+					},
+				},
+				{}
+			),
+			{
+				fromApi: () => ( { posts } ) =>
+					posts.map( post => [ `${ key }-${ post.ID }`, post.views ] ),
+				freshness,
+			}
+		);
+	} );
+};
```

Comment:
I'd like to come in after this PR and perform some optimization here. I don't want to hold it up though. This is a particularly challenging case because of the demands placed on it. This is also one of the reasons the data layer is designed the way it is, so that we can optimize at different layers.

I'm not sure I want to introduce the complexity here but it's worth considering. Imma gonna doodle…

First of all, should we want to consider doing this, it would belong in its own module inside of `data-getters` and not reside in `index.js`

```js
// state/data-getters/recent-post-views.js

// some omitted imports here

export const DEBOUNCE_DELAY = 25; // only needs to allow multiple React elements to request their data

const queue = new Set();

const key = ( siteId, postId ) => …

const requestAndFlush = postIdQueue => {
	requestHttpData( …, { post_ids: [ ...postIdQueue.values() ], num: 30 }, … );
	postIdQueue.clear();
	clearTimeout( postIdQueue.timer );
}

const requestRecentPostViews = postIdQueue => ( siteId, rawPostIds ) => {
	const postIds = Array.isArray( postIds ) ? postIds : [ postIds ];
	const counts = postIds.map( postId => getHttpdata( key( siteId, postId ) ) );
	const toRefresh = counts.filter( o => o.state !== 'pending' && o.lastUpdated < Date.now() - freshness );

	toRefresh.forEach( postId => postIdQueue.add( postId ) );
	
	if ( postIdQueue.size === 100 ) {
		requestAndFlush( postIdQueue );
	} else {
		clearTimeout( postIdQueue.timer );
		postIdQueue.timer = setTimeout( () => requestAndFlush( postIdQueue ), DEBOUNCE_DELAY );
	}

	return counts;
};

export default requestRecentPostViews( queue );
```

What's the value in complicating this one function/module? We can stop making an unexpected call to fetch the list in the post-list and instead rely on individual calls to request the counts _where they are needed_ and yet the data system will perform the optimization to eliminate needless API calls

if we do something like this we'd want some extra people to review it to make sure they are familiar with it and can provide feedback, as well as test it to make sure it does what we expect it to. the behavior should be that batches of 100 post view counts are requested but we have no guarantee on the order. considering that the post list performs a `map` over the posts then we would expect the first 100 to be the first batch.

with this separate module we can add our tests that demonstrate the behaviors as well and guarantee the fundaments are working as we expect them. this is why I baked in the dependency injection when it otherwise wasn't necessary